ui/src/shared/components/FunctionSelector.js

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react'
+import classNames from 'classnames'
+import _ from 'lodash'
+import {INFLUXQL_FUNCTIONS} from 'src/data_explorer/constants'
+
+class FunctionSelector extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      localSelectedItems: this.props.selectedItems,
+    }
+
+    this.onSelect = ::this.onSelect
+    this.onApplyFunctions = ::this.onApplyFunctions
+  }
+
+  onSelect(item, e) {
+    e.stopPropagation()
+
+    const {localSelectedItems} = this.state
+
+    let nextItems
+    if (this.isSelected(item)) {
+      nextItems = localSelectedItems.filter(i => i !== item)
+    } else {
+      nextItems = localSelectedItems.concat(item)
+    }
+
+    this.setState({localSelectedItems: nextItems})
+  }
+
+  isSelected(item) {
+    return this.state.localSelectedItems.indexOf(item) > -1
+  }
+
+  onApplyFunctions(e) {
```

Comment:
since this is where the event is actually handled, i would suggest calling this `handleApplyFunctions`, which you would pass to the `onClick` below. usually there's a series of `on` functions that pass around data until something actually `handle`s it